libs/d3d12/main.c

```diff
@@ -200,26 +225,35 @@ static VkPhysicalDevice d3d12_find_physical_device(struct vkd3d_instance *instan
     free(vk_physical_devices);
     return vk_physical_device;
 }
+#endif
 
 HRESULT WINAPI DLLEXPORT D3D12CreateDevice(IUnknown *adapter, D3D_FEATURE_LEVEL minimum_feature_level,
         REFIID iid, void **device)
 {
     struct vkd3d_instance_create_info instance_create_info;
     PFN_vkGetInstanceProcAddr pfn_vkGetInstanceProcAddr;
     struct vkd3d_device_create_info device_create_info;
-    struct DXGI_ADAPTER_DESC adapter_desc;
     struct vkd3d_instance *instance;
-    IDXGIAdapter *dxgi_adapter;
     HRESULT hr;
 
+#ifdef _WIN32
+    struct DXGI_ADAPTER_DESC adapter_desc;
+    IDXGIAdapter *dxgi_adapter;
+#endif
+
     static const char * const instance_extensions[] =
     {
         VK_KHR_SURFACE_EXTENSION_NAME,
+#ifdef _WIN32
         VK_KHR_WIN32_SURFACE_EXTENSION_NAME,
+#else
+        "VK_KHR_xcb_surface",
```

Comment:
Good enough for now, but definitely worth adding a to-do here... DXVK DXGI uses SDL2 to get the extension to use here, so this will conflict eventually if people force Wayland, etc.

Reply:
(we also probably want to add a bit more flexibility here on the DXVK side too, outside of just SDL2... just haven't had a huge need yet)